src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -22,43 +22,84 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
     initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
         const state = this._store.getState();
         const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+        const paypalOptions = options.paypalCommerce;
 
         if (!paymentMethod.initializationData.clientId) {
             throw new InvalidArgumentError();
         }
 
         return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
             .then(state => {
-                const config = state.config.getStoreConfig();
                 const cart = state.cart.getCart();
 
-                if (!config) {
-                    throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
-                }
-
                 if (!cart) {
                     throw new MissingDataError(MissingDataErrorType.MissingCart);
                 }
 
-                const paramsScript = {
+                const buttonParams: ButtonsOptions = {
+                    createOrder: () => this._setupPayment(cart),
+                    onApprove: data => this._tokenizePayment(paymentMethod.id, data),
+                };
+
+                if (paypalOptions && paypalOptions.style) {
+                    buttonParams.style = this.validateStyleParams(paypalOptions.style);
+                }
+
+                const paramsScript: PaypalCommerceScriptOptions = {
                     clientId: paymentMethod.initializationData.clientId,
-                    currency: config.currency.code,
+                    currency: cart.currency.code,
+                    commit: false,
+                    intent: paymentMethod.initializationData.intent,
                 };
 
+                if (!paymentMethod.initializationData.isPayPalCreditAvailable) {
+                    paramsScript.disableFunding = 'credit';
+                }
+
                 return this._paypalScriptLoader.loadPaypalCommerce(paramsScript)
-                    .then(paypal => {
-                        return paypal.Buttons({
-                            createOrder: () => this._setupPayment(cart),
-                            onApprove: data => this._tokenizePayment(paymentMethod.id, data),
-                        }).render(`#${options.containerId}`);
-                    });
+                    .then(paypal => paypal.Buttons(buttonParams).render(`#${options.containerId}`));
             });
     }
 
     deinitialize(): Promise<void> {
         return Promise.resolve();
     }
 
+    validateStyleParams(style: PaypalButtonStyleOptions): PaypalButtonStyleOptions {
```

Comment:
Probably this should be a private method, 
```
private _validateStyleParams(style: PaypalButtonStyleOptions): PaypalButtonStyleOptions {
```

Reply:
Thank you!